src/components/connectAdvanced.js

```diff
@@ -159,11 +164,14 @@ export default function connectAdvanced(
         // dispatching an action in its componentWillMount, we have to re-run the select and maybe
         // re-render.
         this.subscription.trySubscribe()
-        this.runUpdater()
+        this.updateChildPropsFromReduxStore(false)
       }
 
       shouldComponentUpdate(_, nextState) {
-        return nextState.shouldComponentUpdate
+        if (!connectOptions['pure']) {
```

Comment:
Any particular reason for the bracketed field access here?  Pretty sure you should be able to dot-access it - that would only be an issue if `connectOptions` was undefined, which it won't be.